src/Controls/src/Core/Handlers/Items/iOS/ObservableItemsSource.cs

```diff
@@ -136,18 +136,22 @@ void CollectionChanged(NotifyCollectionChangedEventArgs args)
 					Move(args);
 					break;
 				case NotifyCollectionChangedAction.Reset:
-					Reload();
+					Reload(true);
 					break;
 				default:
 					throw new ArgumentOutOfRangeException();
 			}
 		}
 
-		void Reload()
+		void Reload(bool isDoingReset = false)
 		{
 			var args = new NotifyCollectionChangedEventArgs(NotifyCollectionChangedAction.Reset);
 
-			Count = ItemsCount();
+			//there could be cases where the Reset event doesn't fire before we start adding items if done from a different thread
+			if (isDoingReset)
+				Count = 0;
```

Comment:
"Reset" doesn't mean there are zero items in the collection - it means the contents of the collection have changed dramatically (i.e., a lot).

Reply:
Oh good catch

Reply:
It fixes the issue, because Count is now correct. Right now when getting Reset event when this code was grabbing the Count it was already 13 because items were added .